src/toolkits/style_transfer/style_transfer.cpp

```diff
@@ -616,23 +615,26 @@ void style_transfer::train(gl_sarray style, gl_sarray content,
 
 std::shared_ptr<MLModelWrapper> style_transfer::export_to_coreml(
     std::string filename, std::map<std::string, flexible_type> opts) {
-  flex_int image_width = read_state<flex_int>("image_width");
-  flex_int image_height = read_state<flex_int>("image_height");
-
+  const flex_int image_width = read_opts<flex_int>(opts, "image_width");
+  const flex_int image_height = read_opts<flex_int>(opts, "image_height");
+  const flex_int include_flexible_shape = read_opts<flex_int>(opts, "include_flexible_shape");
+  const flex_string content_feature = read_state<flex_string>("content_feature");
+  const flex_string style_feature = read_state<flex_string>("style_feature");
+  
   flex_dict user_defined_metadata = {
       {"model", read_state<flex_string>("model")},
       {"max_iterations", read_state<flex_int>("max_iterations")},
       {"training_iterations", read_state<flex_int>("training_iterations")},
       {"type", "StyleTransfer"},
-      {"content_feature", read_state<flex_string>("content_feature")},
-      {"style_feature", read_state<flex_string>("style_feature")},
-      {"num_styles", read_state<flex_string>("num_styles")},
+      {"content_feature", content_feature},  // TODO: refactor to take content name and style name
```

Comment:
Not sure what the TODO is trying to communicate. Aren't you already taking `content_feature` and `style_feature`?